Tests for login

onikiforov opened the pull request.

onikiforov committed:
added basic requests and tests for login

bochkala commented on common_requests.py:
```diff
@@ -0,0 +1,41 @@
+
+use_proxy = False
+
+endpoint = "https://python-qa-training.herokuapp.com/api"
```
The naming, in API testing, host + /api would be base_url, and in places where you concatenate URL then: base_url + endpoint (the endpoint e.g '/auth/login').

Other than that no comments, the tests are running smoothly.

onikiforov replied:
done

bochkala commented on test_login.py:
```diff
@@ -0,0 +1,37 @@
+
+
+def test_login_invalid_credentials():
+    password = "123456789"
```
The common practice is to call `global` keyword for the password to avoid variable shadowing: 
```
global password
password = "123456789"
```

More about variable shadowing - https://en.wikipedia.org/wiki/Variable_shadowing

onikiforov replied:
Looks like this will actually set global value of password to "123456789", and I really don't want to do that. I've refactored login tests to use credentials from separate user classes, and thus there will be no need to set password inside the test module code

onikiforov committed:
Refactored login tests, added user data config, refactored requests code

onikiforov committed:
Added parametrized login test in separate module

onikiforov commented:
Refactored login tests:
added user data config to avoid creating parameters in test module
added parametrized login test in separate module
added conftest file with fixtures for future usage

bochkala commented on test_login.py:
```diff
@@ -0,0 +1,37 @@
+    email = ""
+    password = ""
+
+    response = common_requests.login(email, password)
```
@onikiforov just to remind, since you already have `login()` function in `common_requests` you could get rid of local `email` and `password` attributes, e.g passing empty strings directly into the function as follows: `login(email="", password=""). Makes a bit less of code lines, but as for me,  both approaches are ok.
